### PR TITLE
Better workflow

### PR DESCRIPTION
**Pull request by Chocolate4U**
## Description

Current workflow relies on deprecated Node 12 actions which has been out of support since [April 2022](https://github.com/nodejs/Release/#end-of-life-releases). This PR will:
* Update these actions to versions that run on Node 16
* Update Go version to 1.19

## Additional Information

[GitHub Actions: All Actions will begin running on Node16 instead of Node12](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)

### CODE REVIEW

**Comment by Chocolate4U** [comment]
@bootmortis Please review.

**Comment by bootmortis** [comment]
> Copy go.sum from bootmortis/sing-geosite to the repository root to be used as caching dependency file by actions/setup-go@v4 [caching-dependency-files-and-build-outputs](https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs)

We are using two repos that use go, it's not only sing-geosite:
https://github.com/bootmortis/iran-hosted-domains/blob/6c47a632c01473c87d27784b804e4bfbcc67d651/.github/workflows/main.yml#L65

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -16,17 +16,17 @@ jobs:
...
         with:
-          python-version: '3.10.6'
+          python-version: '3.11'
 
       - name: Setup Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v4
         with:
-          go-version: 1.18
+          go-version: 'stable'
```
Comment (bootmortis) [inline .github/workflows/main.yml]:
```suggestion
          go-version: 1.19
```

**Review by bootmortis — CHANGES_REQUESTED** [review]
I'm [not ok with changing the Python version](https://www.youtube.com/watch?v=DgAa08nSyHo).

But about Go, let's stick with `1.19`, ([the version that v2fly uses](https://github.com/v2fly/domain-list-community/blob/8cb0ee0a9ed4587c37cdbee2f937a304ed080a4f/go.mod#L3))

And about caching, we are using two Go-based projects, sing-box, and v2fly, so let's just waste GithHub runners for now.

So let's only stick with these:
- actions/checkout@v3
- actions/setup-python@v4
- actions/setup-go@v4
- go-version: 1.19

And revert other changes.

**Comment by Chocolate4U** [comment]
Ok, I guess "if it works, don't touch it" makes sense :)

About Go, v2fly uses Go v1.19, sing-box uses [latest](https://github.com/bootmortis/sing-geosite/blob/c511eb70dca17ad0c10da3c7c20d7cb94982a2fc/.github/workflows/build.yaml#L15) version of Go, but I think we can stick with 1.19 and update in future if necessary.

About caching, we need to disable it as it is turned on by default, otherwise it will throw a warning nagging about absence of go.sum on every run. then we can get rid of go.sum too.

I will make changes now

**Comment by Chocolate4U** [comment]
Done. Please review.
I have no idea why `requirement.txt` still shows up as changed! Its identical to original
Also I have no idea why 16 commits from previous PR are stalking me here!

**Comment by bootmortis** [comment]
LGTM, THX

**Review by bootmortis — APPROVED** [review]

**Pull request merged**